backend/dataall/modules/worksheets/db/repositories.py

```diff
@@ -0,0 +1,70 @@
+"""
+DAO layer that encapsulates the logic and interaction with the database for worksheets
+"""
+from sqlalchemy import or_, and_
+from sqlalchemy.orm import Query
+
+from dataall.core.group.services.group_resource_manager import GroupResource
+from dataall.db import paginate
+from dataall.modules.worksheets.db.models import Worksheet, WorksheetShare, WorksheetQueryResult
+
+
+class WorksheetRepository(GroupResource):
+    """DAO layer for worksheets"""
+    _DEFAULT_PAGE = 1
+    _DEFAULT_PAGE_SIZE = 10
+
+    def count_resources(self, session, environment, group_uri) -> int:
+        return (
+            session.query(WorksheetQueryResult)
+            .filter(
+                WorksheetQueryResult.AwsAccountId == environment.AwsAccountId
+            )
+            .count()
+        )
+
+    def find_worksheet_by_uri(session, uri) -> Worksheet:
+        return session.query(Worksheet).get(uri)
+    
+    def find_worksheet_share_by_uri(session, uri) -> WorksheetShare:
```

Comment:
the sharing piece is unused right now, we should clean it up

Reply:
done, as agreed I removed the worksheet sharing functionality along the whole repo (BE & FE)